src/stream/src/task/barrier_manager.rs

```diff
@@ -821,14 +851,12 @@ impl LocalBarrierWorker {
         self.add_failure(actor_id, err.clone());
         let root_err = self.try_find_root_failure().await.unwrap(); // always `Some` because we just added one
 
-        let failed_epochs = self.state.epochs_await_on_actor(actor_id).collect_vec();
-        if !failed_epochs.is_empty() {
+        if let Some(actor_state) = self.state.actor_states.get(&actor_id)
+            && !actor_state.inflight_barriers.is_empty()
```

Comment:
Why do we skip reporting the error and reset the stream on `!actor_state.inflight_barriers.is_empty()`? Will we sightly ignore the actor error if the error happens after collecting the previous barrier but before issuing the current barrier?

Reply:
The actors exited with error will be saved in the `failure_actors`, and if the later `inject_barrier` has to collect the barrier from actor in `failure_actors`, we will return with error. 

If we won't check `!actor_state.inflight_barriers.is_empty()` and return the error anyway, we may trigger unnecessary recover for error from actors that are temporarily created in create mview command but has not added to the streaming graph yet.

I just change the current `is_stopping` bool field in `InflightActorState` to a `InflightActorStatus` enum with variants `NotStarted`, `Running` and `Stopping`. We will report the error when `!actor_state.inflight_barriers.is_empty()` or the status is `Running`.

Reply:
> We will report the error when !actor_state.inflight_barriers.is_empty() or the status is Running.

I think you mean when `!actor_state.inflight_barriers.is_empty()` or the status is **not** Running?

Reply:
> > We will report the error when !actor_state.inflight_barriers.is_empty() or the status is Running.
> 
> I think you mean when `!actor_state.inflight_barriers.is_empty()` or the status is **not** Running?

No, I meant `Running`. 

Actor can be of 4 status: 
1. spawned but not added to the streaming graph, which is the `InflightActorStatus::NotStarted`
2. added to the streaming graph, and there are some inflight barriers not collected to it yet, which is `InflightActorStatus::Running`, with non-empty `actor_state.inflight_barriers`
3. added to the streaming graph, and all barriers on it has been collected, no inflight barrier, waiting for injecting next barrier, which is the `InflightActorStatus::Running` with empty `actor_state.inflight_barriers`
4. added to the streaming graph, issued with a stop barrier, waiting for some barriers to be collected, which is `InflightActorStatus::Stopping`

Before this PR, and also at the commit that this comment was on, when we see an actor failure, we ignore it on status 1, report error immediately on status 2 and 4, report error on next inject barrier on status 3.

IIUC, the expected behavior should be, ignore the error on status 1, and report error immediately on status 2, 3 and 4. This difference can be distinguished by `!actor_state.inflight_barriers.is_empty() or the status is Running`. When `!actor_state.inflight_barriers.is_empty()`, the actor must be at status 2 or 4. When the actor has empty inflight barriers, but at `InflightActorStatus::Running`, the actor must be at status 3.

Reply:
> IIUC, the expected behavior should be, ignore the error on status 1, and report error immediately on status 2, 3 and 4. This difference can be distinguished by !actor_state.inflight_barriers.is_empty() or the status is Running. When !actor_state.inflight_barriers.is_empty(), the actor must be at status 2 or 4. When the actor has empty inflight barriers, but at InflightActorStatus::Running, the actor must be at status 3.

Agree. Discussed offline. Previously we handled case 3 incorrectly in the codes and it is fixed in the latest commit.